packages/store/package.json

```diff
@@ -15,9 +15,11 @@
     "codegen": "ts-node ./scripts/codegen.ts",
     "tablegen": "../cli/dist/mud.js tablegen",
     "test": "yarn codegen && forge test",
-    "build": "yarn codegen && rimraf out && forge build && yarn dist && yarn types",
-    "dist": "rimraf abi && mkdir abi && cp -rf out/*.sol/*.json abi/ && yarn rimraf abi/*.metadata.json",
-    "types": "rimraf types && typechain --target=ethers-v5 abi/*.json",
+    "abi": "yarn abi:clean && yarn abi:compile && yarn abi:types",
+    "abi:clean": "rm -rf abi types",
+    "abi:compile": "forge build --out abi --force --skip test script",
+    "abi:types": "typechain --target ethers-v5 \"abi/**/*.sol/!(*.abi).json\" --out-dir types && tsc",
+    "build": "yarn codegen && yarn abi",
```

Comment:
note to self: we can probably remove the `prepare` step since we check in all the necessary build files